src/pages/nickname.tsx

```diff
@@ -1,12 +1,44 @@
 import Image from 'next/image';
-import Link from 'next/link';
-import { useState } from 'react';
+import { useRouter } from 'next/router';
+import { useEffect, useMemo, useState } from 'react';
 
 import Button from '@/components/common/Button';
+import CircleCheckbox from '@/components/common/CircleCheckbox';
+import IconAnchor from '@/components/common/IconAnchor';
 import Input from '@/components/common/Input';
+import { ArrowIcon } from '@/components/icons';
+import useNicknameMutate from '@/hooks/queries/useNicknameMutate';
+
+//FIXME: to setting page
+const agreementList = [
+  { label: '(필수) 서비스 이용약관에 동의', href: '/' },
+  { label: '(필수) 개인정보 처리방침에  동의', href: '/' },
+];
 
 const Nickname = () => {
+  const router = useRouter();
+
   const [name, setName] = useState('');
+  const [agreement, setAgreement] = useState([false, false]);
+
+  const { mutate, isSuccess } = useNicknameMutate();
+
+  const buttonDisabled = useMemo(
+    () => name.length < 2 || name.length > 10 || agreement.some((v) => !v),
+    [name, agreement],
+  );
+
+  const handleAgreementClick = (index: number) => {
+    setAgreement((prev) => (index === 0 ? [!prev[0], prev[1]] : [prev[0], !prev[1]]));
```

Comment:
```suggestion
    setAgreement(([first, second]) => (index === 0 ? [!first, second] : [first, !second]));
```
이런식으로 적으면 좀더 간결해보일것같아요!

Reply:
그렇네요! 수정하겠습니다